### PR TITLE
Add: Benchmark

### PR DESCRIPTION
**Pull request by chris-nee**
### Description
Add benchmarking 

### Changes
- [x] - Readme for benchmarking details
- [x] - Performance Benchmark vector
- [x] - Performance Benchmark string32 string
- [x] - CMakeList build set up
> note for cmakelist, I renamed the string lib to string32 so as to better separate it from utd::string

### CODE REVIEW

**Comment by chris-nee** [comment]
results are stored here - https://drive.google.com/drive/folders/1qENm2qpmgtCSke8k8cRc_80Xde49f_2T?usp=drive_link

given access to your email @benedictkao

**Comment by chris-nee** [comment]
Added a -pfic flag

What does -pFIC flag does ? 
https://stackoverflow.com/questions/966960/what-does-fpic-mean-when-building-a-shared-library

Similar issues in other projects: 
- https://github.com/open-telemetry/opentelemetry-cpp/issues/960#issuecomment-905681929
- https://github.com/google/benchmark/issues/687

File: UTD/CMakeLists.txt [hunk 1]
```diff
@@ -27,7 +27,12 @@ target_sources(timer PRIVATE "src/utils/timer.cpp")
 ### String
 add_library(string SHARED)
 set_target_properties(string PROPERTIES CXX_STANDARD 20)
-target_sources(string PRIVATE "src/data-structures/string32.cpp")
+target_sources(string PRIVATE "src/data-structures/string.cpp")
+
+### String
+add_library(string32 SHARED)
```
Comment (benedictkao) [inline UTD/CMakeLists.txt:33]: Should we just have a single string library instead for simplicity?

Comment (chris-nee) [reply UTD/CMakeLists.txt]: think thats a good suggestion, let me change it

**Review by benedictkao — APPROVED** [review]

**Pull request merged**